src/Extensions/Activities/Internal/ActivityMetricsSender.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Linq;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+using Microsoft.Omex.Extensions.Abstractions.ExecutionContext;
+
+namespace Microsoft.Omex.Extensions.Activities
+{
+	internal sealed class ActivityMetricsSender : IActivitiesEventSender, IDisposable
+	{
+		private readonly Meter m_meter;
+		private readonly Counter<double> m_activityCounter;
+		private readonly Counter<double> m_healthCheckActivityCounter;
+		private readonly IExecutionContext m_context;
+		private readonly IHostEnvironment m_hostEnvironment;
+		private readonly ArrayPool<KeyValuePair<string, object?>> m_arrayPool;
+
+		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment)
+		{
+			m_context = executionContext;
+			m_hostEnvironment = hostEnvironment;
+			m_meter = new Meter("Microsoft.Omex.Activities", "1.0.0");
+			m_activityCounter = m_meter.CreateCounter<double>("Activities");
+			m_healthCheckActivityCounter = m_meter.CreateCounter<double>("HealthCheckActivities");
+			m_arrayPool = ArrayPool<KeyValuePair<string, object?>>.Create();
+		}
+
+		public void SendActivityMetric(Activity activity)
+		{
+			double durationMs = activity.Duration.TotalMilliseconds;
+
+			int tagsCount = s_customTags.Length + activity.TagObjects.Count() + activity.Baggage.Count();
+
+			KeyValuePair<string, object?>[] tags = m_arrayPool.Rent(tagsCount);
+
+			int index = 0;
+
+			foreach (Func<ActivityMetricsSender, Activity, KeyValuePair<string, object?>> getter in s_customTags)
+			{
+				tags[index++] = getter(this, activity);
```

Comment:
it would provide value and then will be incremented, using `++index` would increment and then return.
here demo https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAAwCwAUEQIwUBuEyABBIwLyMwCmA7owNoBdRgG9GxOIyoBfANx0GjWBk4APKmzFzy9JktWENxLfioBOABQQ+etdmwCAlMbOW+dm4UcygA==